server/agents/ReceiptOrchestrator/tools.py

```
@@ -0,0 +1,260 @@
+        Returns:
+            The pass class ID
+        """
+        class_id = f'{issuer_id}.{class_suffix}'
+        
+        # Check if class exists
+        try:
+            self.client.genericclass().get(resourceId=class_id).execute()
+            return class_id
+        except HttpError as e:
+            if e.status_code != 404:
+                raise Exception(f"Error checking class: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)